### ISSUE TITLE
Await does not work with interface extending Promise

### ISSUE THREAD

**Issue by jsedlacek** [Microsoft/TypeScript#12737]
**TypeScript Version:**  2.1.4

**Code**

```ts
interface HelloPromise extends Promise<string> {}

function hello() : HelloPromise {
  return Promise.resolve('hello');
}

async function run() {
  await hello();
};

run();
```

**Expected behavior:** The code is compiled without errors.

**Actual behavior:** Following error is reported:

```
tsc -t es6 --strictNullChecks index.ts
index.ts(8,3): error TS1058: Operand for 'await' does not have a valid callable 'then' member.
```

You can find the repro code here:
https://github.com/jsedlacek/ts-promise-repro

**Comment by jsedlacek** [comment]
Note: This code works in typescript 2.0.4.

**Comment by jwbay** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/12292?

**Comment by mhegazy** [comment]
Fix should be in typescript@next later tonight. Please give it a try and let us know if you are still running into issues.

**Comment by jsedlacek** [comment]
Thanks, I can confirm that typescript@next fixes this issue.
